std/file.d

```diff
@@ -2318,7 +2319,7 @@ private bool ensureDirExists(in char[] pathname)
  * Throws: $(D FileException) on error.
  */
 
-void mkdirRecurse(in char[] pathname)
+void mkdirRecurse()(in char[] pathname)
```

Comment:
I suppose that that depends on whether it affects who can call the code if it's recompiled rather than simply linked. I don't know. I tend to think of the API as being anything in the function signature, whereas the ABI has to do with the resulting binary, and changing the attributes changes the function signature.

Regardless, we routinely make changes that break compatibility in small ways, and the way that D is designed (particularly with attribute inference), you'd have to be pretty disciplined about it to never break any code with changes.